compiler/ilgen/VirtualMachineRegisterInStruct.hpp

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 2016
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#ifndef OMR_VIRTUALMACHINEREGISTERINSTRUCT_INCL
+#define OMR_VIRTUALMACHINEREGISTERINSTRUCT_INCL
+
+#include "ilgen/VirtualMachineRegister.hpp"
+#include "ilgen/IlBuilder.hpp"
+
+namespace OMR
+{
+
+/**
+ * @brief used to represent virtual machine variables that are maintained in a structure stored in a local variable, such as a thread or frame object passed as a parameter to the method
+ *
+ * The value does not need to be a virtual machine register, but often it is the registers
+ * of the virtual machine that are candidates for VirtualMachineRegisterInStruct. An
+ * alternative is VirtualMachineRegister, which can be more convenient if the virtual
+ * machine value is stored in a more arbitrary place or in a structure that isn't readily
+ * accessible inside the compiled method. 
+ * VirtualMachineRegisterInStruct is a subclass of VirtualMachineRegister
+ *
+ * The simulated register value is simply stored in a single local variable, which
+ * gives the compiler visibility to all changes to the register (and enables
+ * optimization / simplification). Because there is just a single local variable,
+ * the Merge() function does not need to do anything (the value is accessible from
+ * the same location at all locations). The Commit() and Reload() functions simply
+ * move the value back and forth between the local variable and the structure that
+ * holds the actual virtual machine state.
+ */
+
+class VirtualMachineRegisterInStruct : public ::OMR::VirtualMachineRegister
```

Comment:
My understanding is that VirtualMachineRegisterInStruct represents a "field" inside some struct whereas VirtualMachineRegister just represents a field loaded from some "static" address. 

If so, is there any key reason why the former extends the latter ? I see that some helper routines like Load and Store are inherited as a convenience because of that hierarchy but could that effect not have been achieved if those routines were in another common superclass that was directly extended by both VirtualMachineRegister and VirtualMachineRegisterInStruct ?